py3status/modules/wifi.py

```diff
@@ -63,16 +64,44 @@ class Py3status:
     cache_timeout = 10
     device = 'wlan0'
     down_color = 'bad'
-    format_down = 'W: down'
-    format_up = 'W: {bitrate} {signal_percent} {ssid}'
+    format = 'W: [{bitrate} {signal_percent} {ssid}|\?show down]'
     round_bitrate = True
     signal_bad = 29
     signal_degraded = 49
     use_sudo = False
 
+    class Meta:
+        deprecated = {
+            'remove': [
+                {
+                    'param': 'format_down',
+                    'msg': 'obsolete parameter use formatter in format',
+                },
+                {
+                    'param': 'format_up',
+                    'msg': 'obsolete parameter use formatter in format',
+                },
+            ],
+        }
+
     def post_config_hook(self):
+        # DEPRECATION WARNING
+        format_down = getattr(self, 'format_down', None)
+        format_up = getattr(self, 'format_up', None)
+
+        if format_down or format_up:
+            self.format = u'{}|{}'.format(
+                format_up or 'W: {bitrate} {signal_percent} {ssid}',
+                format_down or 'W: down',
+            )
+            msg = 'DEPRECATION WARNING: you are using old style configuration '
+            msg += 'parameters you should update to use the new format.'
+            self.py3.log(msg)
```

Comment:
yeah,

if someone has set a format we should use it even if they also supply format_up etc.  that make sense?